internal/core/metadata/rest_deviceprofile.go

```diff
@@ -281,6 +283,16 @@ func restAddProfileByYaml(
 		return
 	}
 
+	err = validateDeviceProfileStruct(dp)
+	if err != nil {
+		if err.Error() == duplicatedCommandNameError {
```

Comment:
Yes, it is fragile, but it is a quick workaround to avoid breaking API backward compatibility.  Do you have any other suggestion?

Your first statement is enough.  We allow the same command name in different profile.
https://github.com/edgexfoundry/go-mod-core-contracts/blob/0746ea1719ff74015da8ef926f06a40e1e317cc2/models/deviceprofile.go#L87-L95

May we update the Swagger file in this PR to the following description?

> If the device profile defines more than one core command with the same name.

Reply:
> Yes, it is fragile, but it is a quick workaround to avoid breaking API backward compatibility. Do you have any other suggestion?

Sure, but we never validated device profiles when using this endpoint...  That said, if this aligns with the `deviceprofile` import using JSON, I guess we can leave it as is. Does our new approach to validation being used for V2 allow more flexible error reporting? I would hate to see us make the same mistake twice...

> Your first statement is enough. We allow the same command name in different profile.
> https://github.com/edgexfoundry/go-mod-core-contracts/blob/0746ea1719ff74015da8ef926f06a40e1e317cc2/models/deviceprofile.go#L87-L95
> 
> May we update the Swagger file in this PR to the following description?
> 
> If the device profile defines more than one core command with the same name.

Yes

Reply:
> Sure, but we never validated device profiles when using this endpoint... That said, if this aligns with the `deviceprofile` import using JSON, I guess we can leave it as is. Does our new approach to validation being used for V2 allow more flexible error reporting? I would hate to see us make the same mistake twice...

We sill avoid this situation in V2

Reply:
> May we update the Swagger file in this PR to the following description?
> If the device profile defines more than one core command with the same name.

Swagger file updated.

Reply:
> Yes, it is fragile, but it is a quick workaround to avoid breaking API backward compatibility. Do you have any other suggestion?

Is it breaking backward compatibility because the code doesn't match our OpenAPI documentation or is it breaking because the non-YAML `deviceprofile` endpoint can return 409, and thus we want the `deviceprofile/upload`, and `../uploadfile` endpoints to do the same???

I would argue that if none of the `deviceprofile` endpoints ever return 409, then it's not breaking backward compatibility if the YAML endpoints don't return it either...

Reply:
So I actually did a bunch of testing this afternoon using Bruce's PR, as well as against the geneva snap (1.2.1). I found that 400 is returned for "duplicate core command names" not 409. In fact, I haven't looked at the actual code path, but even without the fix, the uploadfile endpoint does still return a 400 error in this situation

Reply:
For json format:
I look into the source code
https://github.com/edgexfoundry/edgex-go/blob/master/internal/core/metadata/rest_deviceprofile.go#L74
```
	if err := json.NewDecoder(r.Body).Decode(&dp); err != nil {
		errorHandler.Handle(w, err, errorconcept.Common.InvalidRequest_StatusBadRequest)
		return
	}
```
It only return 400 when uploading the invalid device profile and the implementation not match the API doc.
https://github.com/edgexfoundry/edgex-go/blob/master/api/openapi/v1/core-metadata.yaml#L933